Description: Disables support for Java 6 and 7
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
--- a/src/utils/lombok/javac/Javac.java
+++ b/src/utils/lombok/javac/Javac.java
@@ -432,7 +432,7 @@
 		} catch (IllegalAccessException ignore) {}
 		
 		if (Javac.getJavaCompilerVersion() < 8) {
-			return new JCNoType(((Integer) tag.value).intValue());
+			throw new UnsupportedOperationException("Java before version 8 isn't supported in Debian, please use Lombok binaries from upstream");
 		} else {
 			try {
 				if (CTC_VOID.equals(tag)) {
@@ -452,24 +452,6 @@
 		}
 	}
 	
-	private static class JCNoType extends Type implements NoType {
-		public JCNoType(int tag) {
-			super(tag, null);
-		}
-		
-		@Override
-		public TypeKind getKind() {
-			if (tag == ((Integer) CTC_VOID.value).intValue()) return TypeKind.VOID;
-			if (tag == ((Integer) CTC_NONE.value).intValue()) return TypeKind.NONE;
-			throw new AssertionError("Unexpected tag: " + tag);
-		}
-		
-		@Override
-		public <R, P> R accept(TypeVisitor<R, P> v, P p) {
-			return v.visitNoType(this, p);
-		}
-	}
-	
 	private static final Field JCCOMPILATIONUNIT_ENDPOSITIONS, JCCOMPILATIONUNIT_DOCCOMMENTS;
 	static {
 		Field f = null;
